EESSI-extend-2023.06-easybuild.eb

```diff
@@ -0,0 +1,167 @@
+  if (os.getenv("WORKING_DIR") == nil) then
+    LmodMessage("-- Using /tmp/$USER as a temporary working directory for installations, you can override this by setting the environment variable WORKING_DIR and reloading the module (e.g., /dev/shm is a common option)")
+  end
+end
+working_dir = os.getenv("WORKING_DIR") or pathJoin("/tmp", os.getenv("USER"))
+-- Gather the EPREFIX to use as a sysroot
+sysroot = os.getenv("EESSI_EPREFIX")
+-- Use an installation prefix that we _should_ have write access to
+if (os.getenv("EESSI_CVMFS_INSTALL") ~= nil) then
+  -- Make sure no other EESSI install environment variables are set
+  if ((os.getenv("EESSI_SITE_INSTALL") ~= nil) or (os.getenv("EESSI_PROJECT_INSTALL") ~= nil) or (os.getenv("EESSI_USER_INSTALL") ~= nil)) then
+    LmodError("You cannot use EESSI_CVMFS_INSTALL in combination with any other EESSI_*_INSTALL environment variables")
+  end
+  eessi_cvmfs_install = true
+  easybuild_installpath = os.getenv("EESSI_SOFTWARE_PATH")
+elseif (os.getenv("EESSI_SITE_INSTALL") ~= nil) then
+  -- Make sure no other EESSI install environment variables are set
+  if ((os.getenv("EESSI_PROJECT_INSTALL") ~= nil) or (os.getenv("EESSI_USER_INSTALL") ~= nil)) then
+    LmodError("You cannot use EESSI_SITE_INSTALL in combination with any other EESSI_*_INSTALL environment variables")
+  end
+  easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), 'versions', 'host_injections')
+else
+  -- Deal with user and project installs
+  project_install = os.getenv("EESSI_PROJECT_INSTALL")
+  project_modulepath = nil
+  if (project_install ~= nil) then
+    -- Check the folder exists
+    if not isDir(project_install) then
+      LmodError("The location of EESSI_PROJECT_INSTALL (" .. project_install .. ") does not exist or is not a folder")
+    end
+    if (mode() == "load") then
+      LmodMessage("Configuring for use of EESSI_PROJECT_INSTALL under " .. project_install)
+    end
+    easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), os.getenv("EESSI_CVMFS_REPO"), project_install)
+    project_modulepath = pathJoin(easybuild_installpath, 'modules', 'all')
+  end
+  user_install = os.getenv("EESSI_USER_INSTALL")
+  user_modulepath = nil
+  if (project_install == nil) then
+    if (user_install == nil) then
+      user_install = pathJoin(os.getenv("HOME"), "eessi")
+    else
+      -- Check the folder exists
+      if not isDir(user_install) then
+        LmodError("The location of EESSI_USER_INSTALL (" .. user_install .. ") does not exist or is not a folder")
+      end
+    end
+  end
+  if (user_install ~= nil) then
+    if (mode() == "load") then
+      LmodMessage("Configuring for use of EESSI_USER_INSTALL under " .. user_install)
+    end
+    easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), os.getenv("EESSI_CVMFS_REPO"), user_install)
+    user_modulepath = pathJoin(easybuild_installpath, 'modules', 'all')
+  end
+end
```

Comment:
if both PROJECT and USER are set the code fails to check that user_install is a directory

Reply:
@ocaisa ^

Reply:
Fixed in https://github.com/EESSI/software-layer/pull/371/commits/04c6f0ee56bde553968da02f7daba36cbb363d78